src/include/misc.h

```diff
@@ -333,19 +333,33 @@
 #define WT_STREQ(s, cs) (sizeof(cs) == 2 ? (s)[0] == (cs)[0] && (s)[1] == '\0' : strcmp(s, cs) == 0)
 
 /*
- * Check if a non-empty string matches a byte sequence of len bytes. The macro expects both strings
- * to be of non-zero length in order to be a match. The bytes argument does not need to be
- * null-terminated. Similar to other string related functions, this is not expected to work on NULL
- * values.
+ * Check if a literal string matches the length and content of the supplied bytes/len pair. The
+ * bytes argument does not need to be null-terminated, and it may be null if the supplied length is
+ * zero. Note this is different from the standard strncmp function, which with a supplied zero
+ * length, will always be true.
+ */
+#define WT_STRING_LIT_MATCH(str, bytes, len) \
+    ((len) == strlen("" str "") && strncmp(str, bytes, len) == 0)
+
+/*
+ * Identical to WT_STRING_LIT_MATCH, except that this works with non-literal strings. It is slightly
+ * slower, so WT_STRING_LIT_MATCH is always preferred for literal strings.
  */
 #define WT_STRING_MATCH(str, bytes, len) __wt_string_match(str, bytes, len)
 
 static WT_INLINE bool
 __wt_string_match(const char *str, const char *bytes, size_t len)
 {
-    return (len == 0 || (strncmp(str, bytes, len) == 0 && str[len] == '\0'));
+    return (strncmp(str, bytes, len) == 0 && str[len] == '\0');
 }
 
+/*
+ * CONFIG versions of the WT_STRING_LIT_MATCH and WT_STRING_MATCH macros. These are convenient when
+ * matching WT_CONFIG_ITEMs.
+ */
+#define WT_CONFIG_LIT_MATCH(s, cval) WT_STRING_LIT_MATCH(s, (cval).str, (cval).len)
```

Comment:
I like this macro it makes things easier. We should have done this one a long time ago.